### ISSUE TITLE
[FEATURE] Resource options

### ISSUE THREAD

**Issue by dwisiswant0** [kitabisa/teler#12]
**Is your feature request related to a problem? Please describe.**
In its current state, teler always consumes internet connection if it wants to use resources.

**Describe the solution you'd like**
Give options to downloading resources and storing them locally: in other words the user **DOES NOT** need to use an internet connection if he wants to analyze logs with persistent data.

**Describe alternatives you've considered**
_N/A_

**Additional context**
Like adding an `-dl` or `--download-resources` flag; stores all resources at `$HOME`, checks if all resources are in local storage; otherwise it will use the internet connection to download resources.

**Comment by jtorvald** [comment]
I do think this is a valid feature request. Currently it does not run in production because of firewall restrictions. If there is no reason for downloading resources to analyze logfiles it should be optional I guess. Or is there a reason why this would not be possible @dwisiswant0 ?

**Comment by nse-bse** [comment]
I am also facing this issue (Without Internet, I Can't run teler) not being able to run it in the prod environment. After configuring the whole setup, I now come to know :(. That it requires an internet connection. A lot of resources and time were wasted. :(

**Comment by dwisiswant0** [comment]
> I am also facing this issue (Without Internet, I Can't run teler) not being able to run it in the prod environment. After configuring the whole setup, I now come to know :(. That it requires an internet connection. A lot of resources and time were wasted. :(

Hi @nse-bse, regarding the issue you are experiencing. Dataset resources are only taken once, when you are in a fresh install; it is required, and make sure the `cache` value is set to **true** - then you are free to run teler commands offline.

**Comment by nse-bse** [comment]
Hi, @dwisiswant0, Yes. I tried with the `cache` value set to **true**. But it still asks for an internet connection. I've noticed it is trying to download these files. https://github.com/kitabisa/teler-resources

I confirm all the files are in the `$HOME/.cache/teler-resources` folder. Could you please help resolve this?

**Comment by dwisiswant0** [comment]
@nse-bse - You are right, teler only downloads **once** to run. After that, if the dataset is already in the user-level cache directory, teler will not download it again - if you set the cache `value` to **true**. It is mandatory for the teler engine for dataset update purposes.

If you think it's not as expected. Please open a new issue for better tracking.

**Comment by nse-bse** [comment]
@dwisiswant0 Well, it is still trying to download the files even though the file exists in the cache directory. 

I observed that there is one file in $HOME/.cache/teler-resources `.cached.json`. The timestamp is stored in this JSON file. 

So I am sure that every 12 hours or so, the teler again tries to check for the update. In my prod server, it just worked for one day.  Later, when I saw my SIEM no logs were present. 

eg: `{"updated_at":"2023-01-09T10:14:03Z"}`

![image](https://user-images.githubusercontent.com/33214291/211571675-5108e6ee-42d0-4fee-8455-94f2a660eaa4.png)

Later today, I found the quick fix for this.

I just updated the `.cached.json` with the current timestamp, and it worked without the internet. 

Not sure how the teler handles this; it should just skip updating files when the files are already in the cache directory.
